lib/src/chain.dart

```diff
@@ -82,20 +82,10 @@ class Chain implements StackTrace {
     }
 
     if (!when) {
-      void Function(Object, StackTrace)? newOnError;
-      if (onError != null) {
-        void wrappedOnError(Object error, StackTrace? stackTrace) {
-          onError(
-              error,
-              stackTrace == null
-                  ? Chain.current()
-                  : Chain.forTrace(stackTrace));
-        }
-
-        newOnError = wrappedOnError;
-      }
-
-      return runZoned(callback, onError: newOnError);
+      if (onError == null) return runZoned(callback);
```

Comment:
This seems to disagree with the documentation above which says that the zone must be an error zone if `errorZone` is true, even if `onError` is `null`.

Not sure the previous code did that either. Passing `onError: null` does not make it an error zone.
Maybe remove the documentation? Or the `errorZone` parameter?
(Or at least file an issue to do something).

Reply:
I rewrote the comment to reflect current behavior (as long as I'm reading the code correctly).

I also filed https://github.com/dart-lang/stack_trace/issues/94

I do think it would be good to investigate how much usage this has to see if we can deprecate it.